src/core/prometheus.cc

```diff
@@ -550,9 +553,27 @@ class metric_aggregate_by_labels {
         }
         std::unordered_map<std::map<sstring, sstring>, seastar::metrics::impl::metric_value>::iterator i = _values.find(labels);
         if ( i == _values.end()) {
+            _count.emplace(labels, 1);
             _values.emplace(std::move(labels), m);
         } else {
-            i->second += m;
+            switch (_aggregate_function) {
+            case metrics::aggregate_function_type::SUM:
+                i->second += m;
+                break;
+            case metrics::aggregate_function_type::AVG: {
+                auto count  = _count.find(labels);
```

Comment:
might want to drop the extra space after `count`, or, better off putting something like
```c++
auto& count = _count.find(labels).second;
```